source/common/sequencer_impl.cc

```diff
@@ -135,7 +135,7 @@ void SequencerImpl::run(bool from_periodic_timer) {
     // Re-schedule the periodic timer if it was responsible for waking up this code.
     scheduleRun();
   } else {
-    if (targets_initiated_ == targets_completed_) {
+    if (use_spin_loop_ && (targets_initiated_ == targets_completed_)) {
```

Comment:
Not sure, I follow the logic here; how do we `sleep()` if not spin looping?

Reply:
Ugh, sorry you are right, this was polling instead of sleeping, I added sleep as well. So now we have three options:
- spin
- poll (at Envoy's minimum timer resolution of 1 ms)
- sleep